### DISCUSSION TITLE
Preserving code-server workspace state on DigitalOcean

### DISCUSSION THREAD

**Discussion by litcodeur** [coder/coder#9800] [General] [answered]
Hey there, I've been using a sample template to run a code-server instance on DigitalOcean, and it's been working pretty well. However, I've noticed that when I restart the workspace, all the stuff I had in there from my last session disappears. Is there a way to keep the workspace in the same state as when I last used it, so I can pick up where I left off? Thanks in advance!

**Comment by matifali** [comment, marked as answer]
You should add a persistent disk to your template. For a general overview about persistence in Coder templates see: https://coder.com/docs/v2/latest/templates/resource-persistence.
